lib/jekyll-paginate-v2/generator/paginationModel.rb

```diff
@@ -251,17 +251,23 @@ def paginate(template, config, site_title, all_posts, all_tags, all_categories,
           paginated_page_url = config['permalink']
           first_index_page_url = ""
           if template.data['permalink']
-            first_index_page_url = template.data['permalink']
+            first_index_page_url = File.join(template.data['permalink'],"")
```

Comment:
File.join although it produces the desired result does not make it very clear why it is being used. Please re-consider adding an explicit Util.rb function (there are two others that do similar things to the start of the path). The code will be more readible and easier to understand.